### PR TITLE
feat: adding nes key icon

### PR DESCRIPTION
**Pull request by erickzanardo**
## Status

**READY**

## Description

https://user-images.githubusercontent.com/835641/219022548-1e168c73-d73e-45be-bfc2-711aa61544b3.mov

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: lib/src/widgets/nes_key_icon.dart [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+class NesKeyIcon extends StatelessWidget {
+  /// {@macro nes_key_icon}
+  const NesKeyIcon({
+    super.key,
+    required this.buttonKey,
+    this.pressed = false,
+    this.size,
+  }) : assert(buttonKey.length == 1, 'buttonKey must be a single character');
+
+  /// The key that this icon represents
```
Comment (eliasreis54) [inline lib/src/widgets/nes_key_icon.dart:21]:
```suggestion
  /// The key that this icon represents.
```

File: lib/src/widgets/nes_key_icon.dart [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    return Stack(
+      children: [
+        Positioned(
+          child: NesIcon(
+            size: buttonSize,
+            iconData: pressed
+                ? NesIcons.instance.pressedButton
+                : NesIcons.instance.unpressedButton,
+          ),
```
Comment (eliasreis54) [inline lib/src/widgets/nes_key_icon.dart:55]:
```suggestion
            iconData: iconData
          ),
```

File: lib/src/widgets/nes_icon.dart [hunk 3]
```diff
@@ -142,6 +142,20 @@ class NesIcons {
       '8,8;2,-1;4,0;3,-1;1,0;4,-1;1,0;1,-1;1,0;6,-1;1,0;2,-1;1,0;4,-1;1,0;1,-1;1,0;1,-1;1,0;3,-1;2,0;1,-1;1,0;1,-1;1,0;2,-1;1,0;2,-1;1,0;3,-1;1,0;4,-1;3,0;2,-1',
     ),
   );
+
+  /// An unpressed button icon.
+  late final unpressedButton = NesIconData(
```
Comment (eliasreis54) [inline lib/src/widgets/nes_icon.dart:147]: Just a quick question here, for my understanding... 
Why do you set this as a `late` if you are already initializing it here?

Comment (erickzanardo) [reply lib/src/widgets/nes_icon.dart:147]: by adding the `late` we can use an awesome feature from dart that is the lazy initialization.

So, my code indeed is initializing it, but when this is in runtime, dart will only really initialize the variable when it is accessed for the first time.

So by doing so, we will only allocate the icons in memory that the user is using.

File: lib/src/widgets/nes_key_icon.dart (hunk 2 above)
Comment (erickzanardo) [reply lib/src/widgets/nes_key_icon.dart:54]: good catch!

File: lib/src/widgets/nes_icon.dart (hunk 3 above)
Comment (eliasreis54) [reply lib/src/widgets/nes_icon.dart:147]: Got it! Thanks for explaining! 🤗

**Pull request merged**